### ISSUE TITLE
Text detection model: https://github.com/argman/EAST use with opencv text detectipn dnn

### ISSUE THREAD

**Issue by rawat123** [opencv/opencv#18128]
Hi,

i am using the text detection cpp sample program with provided frozen text detection model .pb file , 
and result detection shares the some common area and after text reorganization it gives text "07" twice.    

  
![detect_result](https://user-images.githubusercontent.com/26247129/90520022-6d123b00-e186-11ea-9ad0-9e8caea0d93b.jpg)

in the image you can see red and green boxes are intersecting.
how can i merge these kind of two intersecting  detection into one.

i tried the https://github.com/argman/EAST#test  eval.py with checkpoint files https://drive.google.com/file/d/0B3APw5BZJ67ETHNPaU9xUkVoV0U/view ,  and this sample gives the ok result with same image, so i tried converting these  checkpoint files to .pb file  with the below code from https://github.com/argman/EAST/issues/140#issuecomment-414336252

````
output_graph_def = tf.graph_util.convert_variables_to_constants(
                sess, # The session is used to retrieve the weights
                tf.get_default_graph().as_graph_def(), # The graph_def is used to retrieve the nodes 
                ["feature_fusion/Conv_7/Sigmoid","feature_fusion/concat_3"]
            ) 

            with tf.gfile.GFile(output_graph, "wb") as f:
                f.write(output_graph_def.SerializeToString())
            print("%d ops in the final graph." % len(output_graph_def.node))
```` 

i created frozen model, but this new frozen model with dnn cpp sample is not working.

please help

**Comment by rawat123** [comment]
i got the answer for my second problem here  https://github.com/opencv/opencv/issues/12491, how to freeze model for opencv dnn text detection sample. 

how is this  https://www.dropbox.com/s/r2ingd0l3zt8hxs/frozen_east_text_detection.tar.gz?dl=1 model is trained , is there any github project.

**Comment by XiaoLaoDi** [comment]
> @rawat123, please read the docs carefully: https://github.com/opencv/opencv/blob/master/samples/dnn/text_detection.cpp
> 
> > Text detection model: https://github.com/argman/EAST
> > Download link: https://www.dropbox.com/s/r2ingd0l3zt8hxs/frozen_east_text_detection.tar.gz?dl=1
> 
> This is a the model from author's repository but with some modifications for correct export.

what does the "some modifications" exactly mean ? I trained the EAST using the EAST repo, and export to pb model, but cv.dnn.readNet() cannot load the model correctly . Do you have any suggestions to export the EAST ckpt to a cv.dnn.readNet readable model ?

**Comment by XiaoLaoDi** [comment]
@rawat123 the following is my ckpt2pb convertion codes.
`
import tensorflow as tf
from tensorflow.python.framework import graph_util
def ckpt2pb():
    with tf.Graph().as_default() as graph_old:
        isess = tf.InteractiveSession()
        ckpt_filename = './tmp/model.ckpt-1011'
        isess.run(tf.global_variables_initializer())
        saver = tf.train.import_meta_graph(ckpt_filename+'.meta', clear_devices=True)
        saver.restore(isess, ckpt_filename)
        constant_graph = graph_util.convert_variables_to_constants(isess, isess.graph_def, ["model_0/feature_fusion/Conv_7/Sigmoid", "model_0/feature_fusion/concat_3"] , variable_names_blacklist="global_step")
        constant_graph = graph_util.remove_training_nodes(constant_graph)
        with tf.gfile.GFile('./tmp/model.pb', mode='wb') as f:
            f.write(constant_graph.SerializeToString())
ckpt2pb()
`

**Comment by XiaoLaoDi** [comment]
@rawat123 and I use the [text_detection.py](https://github.com/opencv/opencv/blob/master/samples/dnn/text_detection.py) to test the trained [EAST](https://github.com/argman/EAST) model.
